osc-server/src/main/java/org/osc/core/broker/util/network/NetworkSettingsApi.java

```diff
@@ -148,4 +90,52 @@ public NetworkSettingsDto getNetworkSettings() {
         return dns;
     }
 
-}
+    private String getNetMask(String ipAddr) {
+        String[] ipAddrParts = ipAddr.split("\\.");
+        String mask = "";
+
+        int octet = Integer.parseInt(ipAddrParts[0]);
+        if (octet <= 127) {
+            mask = "255.0.0.0";
+        } else if (octet >= 128 && octet <= 191) {
+            mask = "255.255.0.0";
+        } else if (octet >= 192 && octet <= 223) {
+            mask = "255.255.255.0";
+        } else if (octet >= 224 && octet <= 239) {
+            mask = "255.0.0.0";
+        } else if (octet >= 240 && octet <= 254) {
+            mask = "255.0.0.0";
+        }
+    return mask;
+    }
+
+    private String getDefaultGateway() {
```

Comment:
why private method? you can directly use getNetworkDefaultGeteway

Reply:
The intention was to make this method generic with public static but time being i merged as suggested.